src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/LocationAttribute.kt

```diff
@@ -107,32 +105,30 @@ class LocationAttribute(
           }
         }
       }
-      LocationUsage.SCHEDULE -> locationSchedule.check(probationTeam, startingOnDateTime)
+      LocationUsage.SCHEDULE -> getScheduleAvailability(probationTeam, startingOnDateTime)
       else -> return AvailabilityStatus.NONE
     }
   }
 
-  private fun List<LocationSchedule>.check(probationTeam: ProbationTeam, startingOnDateTime: LocalDateTime): AvailabilityStatus = filter {
-    startingOnDateTime.dayOfWeek.between(it.startDayOfWeek, it.endDayOfWeek) && startingOnDateTime.toLocalTime().isBetween(it.startTime, it.endTime)
-  }
-    .let {
-      if (it.isEmpty()) {
-        return AvailabilityStatus.SHARED
-      } else {
-        if (any { schedule ->
-            schedule.isAvailableForProbationTeam(
-              probationTeam,
-              startingOnDateTime,
-            )
-          }
-        ) {
-          return AvailabilityStatus.PROBATION_TEAM
-        }
-        if (any { schedule -> schedule.isAvailableForAnyProbationTeam(startingOnDateTime) }) return AvailabilityStatus.PROBATION_ANY
-        return AvailabilityStatus.NONE
-      }
+  /**
+   * We need to look at the schedule as a whole to determine availability. A schedule on its own is not enough.
+   */
+  private fun getScheduleAvailability(probationTeam: ProbationTeam, dateAndTime: LocalDateTime): AvailabilityStatus {
+    val schedules = locationSchedule.filter { it.fallsOn(dateAndTime) }
+
+    // If there aren't any schedules which fall on the requested date and time then the default fallback is SHARED
+    if (schedules.isEmpty()) {
+      return AvailabilityStatus.SHARED
+    } else {
+      // If there are matches then look to see if any match the following
```

Comment:
We know at this point we only have schedules that fall within the time window of interest given the above ergo we only need to check if the following apply here.